pkg/helm/client/client.go

```diff
@@ -98,16 +101,21 @@ func NewRESTClientGetter(mgr manager.Manager, ns string) (genericclioptions.REST
 
 var _ kube.Interface = &ownerRefInjectingClient{}
 
-func NewOwnerRefInjectingClient(base kube.Client, ownerRef metav1.OwnerReference) kube.Interface {
+func NewOwnerRefInjectingClient(base kube.Client, ownerRef metav1.OwnerReference,
+	mgr manager.Manager, cr *unstructured.Unstructured) kube.Interface {
 	return &ownerRefInjectingClient{
-		refs:   []metav1.OwnerReference{ownerRef},
-		Client: base,
+		refs:       []metav1.OwnerReference{ownerRef},
```

Comment:
We can derive the owner reference from `cr`, so I think we remove that field from the struct and create the owner reference on the fly when we need it in `Build`